modules/apps/frontend-js/frontend-js-walkthrough-web/src/main/resources/META-INF/resources/useLocalStorage.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useCallback, useState} from 'react';
+
+export function useLocalStorage(key, initialValue) {
```

Comment:
Not to be done at this PR but... :p

This hook is interesting and we might move it to *react-web module and then replace it on modules. I know, I'm creating chaos but it complains DRY.

Reply:
Well, I'm not sure why this is a very specific scenario for this component where we are storing user state on the client-side instead of the backend, at least I don't remember use cases in DXP that need localStorage to persist states.

Reply:
we may work on this in the context of cookie manager as localstorage usage has to be controlled in some way according to user consent. So we can leave that here then see if we need a different API to "safely" access the localstorage from DXP code, then rewire this later on